spec/reek/smells/too_many_constants_spec.rb

```diff
@@ -0,0 +1,174 @@
+require_relative '../../spec_helper'
+require_lib 'reek/smells/too_many_constants'
+require_relative 'smell_detector_shared'
+
+RSpec.describe Reek::Smells::TooManyConstants do
+  context 'counting constants' do
+    it 'should not report for non-excessive constants' do
+      src = <<-EOS
+        class Dummy
+          A = B = C = D = E = 1
+        end
+      EOS
+
+      expect(src).not_to reek_of(:TooManyConstants)
+    end
+
+    it 'should not account class definition' do
+      src = <<-EOS
+        module Dummiest
+          class Dummy
+            A = B = C = D = E = 1
+
+            Error = Class.new(StandardError)
+          end
+        end
+      EOS
+
+      expect(src).not_to reek_of(:TooManyConstants)
+    end
+
+    it 'should not account struct definition' do
+      src = <<-EOS
+        module Dummiest
+          class Dummy
+            A = B = C = D = E = 1
+
+            Struct = Struct.new
+          end
+        end
+      EOS
+
+      expect(src).to_not reek_of(:TooManyConstants)
+    end
+
+    it 'should count each constant only once' do
+      src = <<-EOS
+        class Good
+          A = B = C = D = E = 1
+        end
+
+        class Bad
+          A = B = C = D = E = 1
+        end
+
+        class Ugly
+          A = B = C = D = E = 1
+        end
+      EOS
+
+      expect(src).not_to reek_of(:TooManyConstants)
+    end
+
+    it 'should count each constant only once for each class' do
+      src = <<-EOS
+        module Movie
+          class Good
+            A = B = C = D = E = 1
+          end
+
+          class Bad
+            F = G = H = I = J = 1
+          end
+
+          class Ugly
+            K = L = M = N = O = 1
+          end
+        end
+      EOS
+
+      expect(src).not_to reek_of(:TooManyConstants)
+    end
+
+    it 'should count each constant only once for each namespace' do
+      src = <<-EOS
+        module Movie
+          A = B = C = D = E = 1
+
+          class Good
+            F = 1
+          end
+        end
+      EOS
+
+      expect(src).not_to reek_of(:TooManyConstants)
+    end
+
+    it 'should report for excessive constants inside a class' do
+      src = <<-EOS
+        class Dummy
+          A = B = C = D = E = 1
+          F = 1
+        end
+      EOS
+
+      expect(src).to reek_of(:TooManyConstants)
+    end
+
+    it 'should report for excessive constants inside a module' do
+      src = <<-EOS
+        module Dummiest
+          A = B = C = D = E = 1
+          F = 1
+
+          class Dummy
+          end
+        end
+      EOS
+
+      expect(src).to reek_of(:TooManyConstants, context: 'Dummiest')
+    end
+  end
+
+  context 'smell report' do
```

Comment:
@mvz I'd rather repeat it inside the `it` block as many times as needed instead of using this approach or `let`. The reason behind it is even though repeating it goes against DRY principle, in an `it` block I will have everything that is required to understand what is being tested and how.

For instance, let's suppose we start irresponsibly adding more `context`s and `it`s blocks inside this context. Even though I know there's a `src` it will become more and more difficult to find where it is defined. In every line addition the `src` variable will turn into a even more mysterious guest.

Even though the big block of text I'm fine with any possible conclusion. What do you think?